src/mds/MDLog.cc

```diff
@@ -604,11 +614,33 @@ void MDLog::try_to_commit_open_file_table(uint64_t last_seq)
   }
 }
 
-void MDLog::trim(int m)
+void MDLog::log_trim_upkeep(void) {
+  dout(10) << dendl;
+
+  submit_mutex.lock();
+  while (!upkeep_log_trim_shutdown.load()) {
+    {
+      // adjust lock order
+      submit_mutex.unlock();
+      std::scoped_lock mlock(mds->mds_lock);
+      submit_mutex.lock();
+
+      if (mds->is_active() || mds->is_stopping()) {
+	trim();
+      }
```

Comment:
do you think adding a log line dout(10) << "Cannot trim since the mds is not active or stopping" would be helpful?

Reply:
Not really.